src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -92,8 +92,15 @@ public function read($key)
         $this->ensureBucketExists();
         $options = $this->getOptions($key);
 
+        // Copy ContentType metadata to local metadata
+        $object = $this->service->getObject($options);
```

Comment:
Could you instead wrap this line into the try/catch block, rather than the `$object->get('Body')`.